src/scripting_api/app.js

```diff
@@ -475,8 +485,18 @@ class App extends PDFObject {
   }
 
   execMenuItem(item) {
+    if (!this._document.obj._userActivation) {
+      return;
+    }
+    this._document.obj._userActivation = false;
+
     switch (item) {
       case "SaveAs":
+        if (this._document.obj._disableSaving) {
+          return;
+        }
+        this._send({ command: item });
+        break;
```

Comment:
Don't we need similar code in the `"Print"` case below?

Reply:
There are no zoom events after having triggered a zoom so it doesn't make sense.
FYI, save/print trigger a `Will***` and `Did***`, and I'm just disabling save/print when the associated callback are called because those events aren't trigger by the user themselves (and because it doesn't make sense, but I can change my opinion if someone has a good example).